rmc-docs/src/tutorial-first-steps.md

```diff
@@ -0,0 +1,196 @@
+But RMC still catches the issue:
+
+```
+** Results:
+./src/lib.rs function estimate_size
+[estimate_size.pointer_dereference.1] line 10 dereference failure: pointer NULL in *var_10: FAILURE
+
+** 1 of 1 failed (2 iterations)
+VERIFICATION FAILED
+```
+
+**Can you find an example where Rust will not complain, and RMC will?**
+
+<details>
+<summary>Click to show one possible answer</summary>
+
+```
+return 1 << x;
+```
+
+Overflow (addition, multiplication, etc, and this case, [bitshifting by too much](https://github.com/rust-lang/rust/issues/10183)) is also caught by RMC:
+
+```
+** Results:
+./src/lib.rs function estimate_size
+[estimate_size.assertion.1] line 10 attempt to shift left by `move _10`, which would overflow: FAILURE
+[estimate_size.undefined-shift.1] line 10 shift distance too large in 1 << var_10: FAILURE
+
+** 2 of 2 failed (2 iterations)
+VERIFICATION FAILED
+```
+
+</details>
+
+## Assertions, Assumptions, and Harnesses
+
+It seems a bit odd that we can take billions of inputs, but our function clearly only handles up to a few thousand.
+Let's codify this fact about our function by asserting some reasonable bound on our input, after we've fixed our bug:
+
+```rust
+{{#include tutorial/rmc-first-steps/tests/final-form.rs:code}}
+```
+
+Now we have stated our previously implicit expectation: this function should never be called with inputs that are too big.
+But if we attempt to verify this, we get a problem:
+
+```
+** Results:
+./tests/final-form.rs function estimate_size
+[estimate_size.assertion.1] line 3 assertion failed: x < 4096: FAILURE
+
+** 1 of 1 failed (2 iterations)
+VERIFICATION FAILED
+```
+
+We intended this to be a precondition of calling the function, but RMC is treating it like a failure.
+If we call this function with too large of a value, it will crash with an assertion failure.
+But we know that, that was our intention.
+
+This is the purpose of _proof harnesses_.
+Much like property testing (which would also find this assertion failure as a bug), we need to set up our preconditions, call the function in question, then assert our post conditions.
+Here's a revised example of the proof harness, one that now succeeds:
+
+```rust
+{{#include tutorial/rmc-first-steps/tests/final-form.rs:rmc}}
+```
+
+But now we must wonder if we've really fully tested our function.
+What if we revise the function, but forget to update the assumption in our proof harness to cover the new range of inputs?
+
+Fortunately, RMC is able to report a coverage metric for each proof harness.
+Try running:
+
+```
+rmc --visualize tests/final-form.rs
+open report/html/index.html
+```
+
+The beginning of the report includes coverage information.
+Clicking through to the file will show fully-covered lines in green.
+Lines not covered by our proof harness will show in red.
+
+1. Try changing the assumption in the proof harness to `x < 2048`. Now the harness won't be testing all posssible cases.
```

Comment:
An extra 's' in "possible".

Reply:
I really need to get spell check working in vs code :/